src/board/board_tests.rs

```diff
@@ -0,0 +1,255 @@
+#[cfg(test)]
+use crate::board::*;
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    #[test]
+    fn test_starting_board_from_str() {
+        let board_from_fen =
+            Board::from_str("rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR").unwrap();
+        let board_hardcoded = starting_position();
+        assert_eq!(board_from_fen, board_hardcoded);
+    }
+
+    #[test]
+    fn test_error_fen_string() {
+        let board_from_fen =
+            Board::from_str("rnbqkbna/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR").unwrap_err();
+        assert_eq!(
+            board_from_fen,
+            ParseFenError::InvalidPiecePlacement("a".to_string())
+        );
+    }
+}
+
+#[cfg(test)]
```

Comment:
remove

Reply:
Removing this results in a warning for dead code. what is the proper way of dealing with test only functions?